sklearn/base.py

```diff
@@ -273,6 +273,8 @@ def __repr__(self, N_CHAR_MAX=700):
     def __getstate__(self):
         try:
             state = super().__getstate__()
+            if state is None:
+                state = self.__dict__.copy()
```

Comment:
python seems to be making a distinction between a `__dict__` attached to the instance, and a `__dict__` attached to the class. I'm not sure if we should do the same:

```python
>>> class A:
...     __dict__ = {"a": 10}
... 
>>> a = A()
>>> a.__dict__
{'a': 10}
>>> a.__getstate__()
>>> a.__dict__.update({"b": 3})
>>> a.__dict__
{'a': 10, 'b': 3}
>>> a.__getstate__()
>>> class B:
...     ...
... 
>>> b = B()
>>> b.b = 10
>>> b.__dict__
{'b': 10}
>>> b.__getstate__()
{'b': 10}
```

Reply:
Having a `__dict__` as class attribute results in some (for me) quite surprising behavior. Not sure if that should be supported and if yes, how to differentiate that here.

```python
class F:
    __dict__ = {"x": 0}
    def __init__(self):
        self.y = 1

f = F()
print(f"{f.__getstate__()=}")  # {'y': 1}
print(f"{f.__dict__=}")  # {'x': 0}
print(f"{'x' in dir(f)=}")  # True
print(f"{'y' in dir(f)=}")  # False
f.y  # works
f.x  # AttributeError:  'F' object has no attribute 'x'
```

Reply:
> python seems to be making a distinction between a `__dict__` attached to the instance, and a `__dict__` attached to the class.

Class attributes and instance attributes are different, which is why you see what you are seeing. While this is maybe not a super useful answer, as it is just stating a fact about how Python works, I don't understand what you are trying to do that leads you down the path of poking around at this level. Can you give an example or use-case? Would help with giving a more helpful answer.